src/optimizer/rule_query_rewrite.cpp

```diff
@@ -0,0 +1,191 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// transitive_predicates.cpp
+//
+// Identification: src/optimizer/rule_query_rewrite.cpp
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <memory>
+
+#include "catalog/column_catalog.h"
+#include "catalog/index_catalog.h"
+#include "catalog/table_catalog.h"
+#include "optimizer/rule_query_rewrite.h"
+#include "optimizer/operators.h"
+#include "optimizer/optimizer_metadata.h"
+#include "optimizer/properties.h"
+#include "optimizer/rule_impls.h"
+#include "optimizer/util.h"
+
+namespace peloton {
+namespace optimizer {
+
+//===--------------------------------------------------------------------===//
+// Transitive predicates rules
+//===--------------------------------------------------------------------===//
+
+///////////////////////////////////////////////////////////////////////////////
+/// LogicalGet
+TransitivePredicatesLogicalGet::TransitivePredicatesLogicalGet() {
+  type_ = RuleType::TRANSITIVE_PREDICATES_LOGICAL_GET;
+
+  match_pattern = std::make_shared<Pattern>(OpType::Get);
+}
+
+bool TransitivePredicatesLogicalGet::Check(std::shared_ptr<OperatorExpression> input,
```

Comment:
add a TODO for what you actually want to do here. same for the other Check methods that do this